### ISSUE TITLE
Filtering an array using `String` native coercion function fails.

### ISSUE THREAD

**Issue by nohehf** [sindresorhus/eslint-plugin-unicorn#2089]
When having to filter an array using the native `String` function like this:
```TS
let array = // whatever array
let filtered = array.filter(String);
```
I get the `unicorn/no-array-callback-reference` error.

If I then change it to comply with this rule:
```TS
let array = // whatever array
let filtered = array.filter((s) => String(s));
```

I get `unicorn/prefer-native-coercion-functions`.

I believe String should pass the [no-array-callback-reference rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md), as `Boolean` does.

**Comment by fisker** [comment]
I don't understand what kind of array are you filtering. Why don't you use `.filter(s => s != '')`?

**Comment by fisker** [comment]
It seems [people actually use this pattern](https://github.com/search?q=.filter%28String%29&type=code).
Most of these cases are array of strings, i still think `.filter(s => s != '')` is more clear, and `.filter(Boolean)` also work but more common.

**Comment by nohehf** [comment]
> I don't understand what kind of array are you filtering. Why don't you use `.filter(s => s != '')`? Can you share a real world case?

I can have nulls, so `.filter(s => s != '')` is not working. 

> It seems [people actually use this pattern](https://github.com/search?q=.filter%28String%29&type=code). Most of these cases are array of strings, i still think `.filter(s => s != '')` is more clear, and `.filter(Boolean)` also work but more common.

After testing, I agree that `.filter(Boolean)` works well in my use case.

The only use case I can imagine would be something like this, but I see no real-world use for it:
```TS
const array = [1, "2", 0, "0", ""]
const boleanFilteredArray = array.filter(Boolean)
// boleanFilteredArray: [ 1, '2', '0' ]
const stringFilteredArray = array.filter(String)
// stringFilteredArray: [ 1, '2', 0, '0' ]
```
My bad then for misusing `.filter(String)`

However, `.map(String)` is useful, but seems to be ignored from the `unicorn/no-array-callback-reference` rule, so there might be no issue unless, in some cases, `.filter(String)` is necessary.

**Comment by fisker** [comment]
> I can have null

But `String(null)` and `String(undefined)` are all truly. The result should be same as `.filter(String)`.
